libflux/src/core/semantic/types.rs

```diff
@@ -153,16 +155,20 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::CannotUnify(t, with) => write!(f, "{} != {}", t, with),
-            Error::CannotConstrain(t, with) => write!(f, "{} is not {}", t, with),
+            Error::CannotUnify(expected, found) => write!(f, "{} != {}", found, expected),
```

Comment:
Should we change this error message too?

Instead of saying `int != string` is there better way to communicate what is going on now that we know which type is expected and which one was found?

For example something like `found type 'int' but was expecting 'string'`?

I know its slightly unrelated to the issue being fixed but it all we have to change is this one line know I think we should add it.

Reply:
How does `expected int but found string` read?

Reply:
I like it, short and simple. Hopefully with the context of where the error occurs it will lead users in the right direction.